File: Second year/Design and Programming Environments/TicketManagerCSharp/TicketManagerCSharp/repository/EventRepository.cs
using log4net;
using System;
using System.Collections.Generic;
using System.Data;
using TicketManagerCSharp.domain;
using TicketManagerCSharp.utils;

namespace TicketManagerCSharp.repository
{
    public class EventRepository : IRepository<int, Event>
    {
        private static readonly ILog log = LogManager.GetLogger("EventRepository");

        private IDictionary<String, string> props;

        public EventRepository(IDictionary<String, string> props)
        {
            log.Info("Creating EventRepository...");
            this.props = props;
        }

        public void Delete(int id)
        {
            log.InfoFormat("Entering Delete with new value {0}...", id);
            IDbConnection con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from Events where id=@id";
                IDbDataParameter idParam = comm.CreateParameter();
                idParam.ParameterName = "@id";
                idParam.Value = id;
                comm.Parameters.Add(idParam);
                var rez = comm.ExecuteNonQuery();
                if (rez == 0)
                {
                    throw new RepositoryException("Error:Can't delete Event!");
                }
                log.InfoFormat("Exiting Delete");
            }
        }

        public IEnumerable<Event> FindAll()
        {
            IDbConnection con = DBUtils.getConnection(props);
            log.InfoFormat("Entering findAll...");
            IList<Event> events = new List<Event>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select id,game_name,game_price,free_seats from Events";
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int idE = dataR.GetInt32(0);
                        string name = dataR.GetString(1);
                        double gamePrice = dataR.GetDouble(2);
                        int seats = dataR.GetInt32(3);
                        Event e = new Event(idE, name, gamePrice, seats);
                        events.Add(e);
                    }
                }
            }
            log.InfoFormat("Exiting findAll");
            return events;
        }

        public Event FindOne(int id)
        {
            log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select id,game_name,game_price,free_seats from Events where id=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int idE = dataR.GetInt32(0);
                        string name = dataR.GetString(1);
                        double gamePrice = dataR.GetDouble(2);
                        int seats = dataR.GetInt32(3);
                        Event e = new Event(idE, name, gamePrice, seats);
                        log.InfoFormat("Exiting findOne with value{0}", e);
                        return e;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public void Save(Event entity)
        {
            log.InfoFormat("Entering Save with new value {0}...", entity);
            var con = DBUtils.getConnection(props);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "insert into Events(game_name,game_price,free_seats) values (@game_name,@game_price,@free_seats)";
                var nameParam = comm.CreateParameter();
                nameParam.ParameterName = "@game_name";
                nameParam.Value = entity.GameName;
                comm.Parameters.Add(nameParam);

                var gamePriceParam = comm.CreateParameter();
                gamePriceParam.ParameterName = "@game_price";
                gamePriceParam.Value = entity.GamePrice;
                comm.Parameters.Add(gamePriceParam);

                var freeSeatsParam = comm.CreateParameter();
                freeSeatsParam.ParameterName = "@free_seats";
                freeSeatsParam.Value = entity.FreeSeats;
                comm.Parameters.Add(freeSeatsParam);

                var result = comm.ExecuteNonQuery();
                if (result == 0)
                {
                    throw new RepositoryException("Error:Can't add event!");
                }
                log.InfoFormat("Exiting Save");
            }
        }

        public int Size()
        {
            log.InfoFormat("Calculating Events database Size...");
            IDbConnection con = DBUtils.getConnection(props);
            int size = -1;
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select count(*) from Events";

                var rez = comm.ExecuteNonQuery();
                if (rez == 0)
                {
                    throw new RepositoryException("Error:Can't calculate size!");
                }

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        size = dataR.GetInt32(0);
                    }
                }
            }
            log.InfoFormat("Finished calculation!");
            return size;
        }

        public void Update(int id, Event entity)
        {
            log.InfoFormat("Entering Update with value {0}", id);
            var con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "update Events set game_name=@game_name,game_price=@game_price,free_seats=@free_seats where id=@id";
                var nameParam = comm.CreateParameter();
                nameParam.ParameterName = "@game_name";
                nameParam.Value = entity.GameName;
                comm.Parameters.Add(nameParam);

                var gamePriceParam = comm.CreateParameter();
                gamePriceParam.ParameterName = "@game_price";
                gamePriceParam.Value = entity.GamePrice;
                comm.Parameters.Add(gamePriceParam);

                var freeSeatsParam = comm.CreateParameter();
                freeSeatsParam.ParameterName = "@free_seats";
                freeSeatsParam.Value = entity.FreeSeats;
                comm.Parameters.Add(freeSeatsParam);

                var idParam = comm.CreateParameter();
                idParam.ParameterName = "@id";
                idParam.Value = id;
                comm.Parameters.Add(idParam);

                var result = comm.ExecuteNonQuery();
                if (result == 0)
                {
                    throw new RepositoryException("Error:Can't update Event!");
                }
                log.InfoFormat("Exiting Update");
            }
        }
    }
}

File: Second year/Design and Programming Environments/TicketManagerJava/Persistance/src/main/java/Repository/EventRepository.java
package Repository;

import ModelDomain.Event;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class EventRepository implements IRepository<Integer, Event> {
    private Connection connection;

    public EventRepository() {
        connection = DatabaseConnection.getConnection();
    }

    @Override
    public int size() {

        try (PreparedStatement preStmt = connection.prepareStatement("select count(*) as [SIZE] from Events")) {
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    return result.getInt("SIZE");
                }
            }
        } catch (SQLException e) {
            System.out.println("Error DB ModelDomain.Event Size " + e);
        }
        return 0;
    }

    @Override
    public void save(Event entity) {
        try (PreparedStatement preStmt = connection.prepareStatement("insert into Events values (?,?,?,?)")) {
            preStmt.setString(1, entity.getGameName());
            preStmt.setDouble(2, entity.getGamePrice());
            preStmt.setInt(3, entity.getFreeSeats());
            preStmt.setString(4, entity.getGameState().toString());
            int result = preStmt.executeUpdate();
        } catch (SQLException e) {
            System.out.println("Error DB ModelDomain.Event Save " + e);
        }
    }

    @Override
    public void delete(Integer integer) {
        try (PreparedStatement preStmt = connection.prepareStatement("delete from Events where id=?")) {
            preStmt.setInt(1, integer);
            int result = preStmt.executeUpdate();
        } catch (SQLException e) {
            System.out.println("Error DB ModelDomain.Event Delete " + e);
        }
    }

    @Override
    public void update(Integer integer, Event entity) {
        try (PreparedStatement preStmt = connection.prepareStatement("update Events set game_name=? ,game_price=?, free_seats=? ,game_state=? where id=?")) {
            preStmt.setString(1, entity.getGameName());
            preStmt.setDouble(2, entity.getGamePrice());
            preStmt.setInt(3, entity.getFreeSeats());
            preStmt.setString(4, entity.getGameState().toString());
            preStmt.setInt(5, integer);
            int result = preStmt.executeUpdate();

        } catch (SQLException ex) {
            System.out.println("Error DB " + ex);
        }
    }

    @Override
    public Event findOne(Integer integer) {
        try (PreparedStatement preStmt = connection.prepareStatement("select * from Events where id=?")) {
            preStmt.setInt(1, integer);
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    int id = result.getInt("id");
                    String gameName = result.getString("game_name");
                    double price = result.getDouble("game_price");
                    int seats = result.getInt("free_seats");
                    Event event = new Event(id, gameName, price, seats);
                    return event;
                }
            }
        } catch (SQLException e) {
            System.out.println("Error DB ModelDomain.Event FindOne " + e);
        }
        return null;
    }

    @Override
    public Iterable<Event> findAll() {
        List<Event> events = new ArrayList<>();
        try (PreparedStatement preStmt = connection.prepareStatement("select * from events")) {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    int id = result.getInt("id");
                    String gameName = result.getString("game_name");
                    double price = result.getDouble("game_price");
                    int seats = result.getInt("free_seats");
                    Event event = new Event(id, gameName, price, seats);
                    events.add(event);
                }
            }
        } catch (SQLException e) {
            System.out.println("Error DB ModelDomain.Event FindAll " + e);
        }
        return events;
    }
}
